package.json

```diff
@@ -67,6 +67,7 @@
     "grunt-mocha-test": "^0.13.3",
     "grunt-newer": "^1.1.0",
     "grunt-simple-nyc": "^3.0.1",
+    "grunt-babel": "^8.0.0",
```

Comment:
# Main problem
To understand the main problem, let me explicitly state it here. The current build pipeline takes all the code and runs `browserify` on it, which makes a single file, and that is passed on to `uglify`, which makes it very small, browser ready.


`Uglify` does not understand ES6 code, so it needs to be transpiled before it gets here. This is `browserify's` (and more specifically,` babelify's`) job. `Browserify` can be coupled with babelify to transpile between javascript versions before compiling and bundling. 

# Babelify and options
Inside the `tasks/build/browserify.js` file we have, among other things, this important variable:

```javascript
const babelifyOpts = {
		// add some options
        plugins: ['static-fs']
      };
```

which lets us modify the behaviour of `babelify`. As of right now, it only has a plugin called `static-fs`, which converts all asynchronous file system calls to their "hardcoded" version, statically inlining these calls.

We can pass in more options to change the behaviour:

```javascript
const babelifyOpts = {
		// add some options
        plugins: ['static-fs'],
        presets: ['@babel/preset-env'],
        global: true,
      };
```

This uses the `present-env` preset, recommended for our use case and the `global` option tells `babelify` to also process and transpile code coming from `node_modules`. The default behavior does not do that.

# No solution found

After playing with every combination, some problem arises.

- If we don't include `global`, we cannot have `node_modules` transpiling so future dependencies from other contributors that also work similarly to `gifenc` won't work during build time.
- If we don't include the `static-fs`, `mochaChrome` breaks:
```
Running "mochaChrome:yui" (mochaChrome) task
p5.js translator called before translations were loaded
p5.js translator called before translations were loaded
[stuck here forever]
```
- Including `static-fs` AND `global` breaks the `static-fs` since it cannot compile other libraries (like `opentype`) with their file calls.
```
Fatal error: /Users/jesi/repositorios/p5.js/node_modules/opentype.js/dist/opentype.js: Not able to statically evaluate the expression(s) for babel-plugin-static-fs.
Try changing your source code to something that can be evaluated at build-time, e.g.
    const src = fs.readFileSync(__dirname + '/foo.txt', 'utf8');
 while parsing file: /Users/jesi/repositorios/p5.js/node_modules/opentype.js/dist/opentype.js
```

# My initial solution
Initially, I simply included the `grunt-babel` option since the main problem was I could not get `Uglify` to work. This worked fine immediately, albeit not knowing why and not knowing that `babelify` was already present.

For what it's worth, the `grunt-babel` has not given absolutely any problem, since it is set after the `browserify` task which means it gets to transpile the already bundled app. I believe this can open a door for future contributors that won't have to deal with this topic but just develop a better experience for p5.js.

If this is not enough to justify the `grunt-babel` inclusion, my only guess is that I should not include `gifenc` at all and figure out another way of using that code. Totally open for discussion!